Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeLibsAPI.cs

```diff
@@ -0,0 +1,1006 @@
+        public static string GetDocumentClassControlType(IVBProject project, string documentClassName, string controlName)
+        {
+            using (var typeLib = TypeLibWrapper.FromVBProject(project))
+            {
+                return GetDocumentClassControlType(typeLib, documentClassName, controlName);
+            }
+        }
+
+        /// <summary>
+        /// Returns the class progID of a control on a UserForm
+        /// </summary>
+        /// <param name="projectTypeLib">Low-level ITypeLib wrapper representing the VBA project</param>
+        /// <param name="documentClassName">Document class name, as declared in the VBA project</param>
+        /// <param name="controlName">Control name, as declared on the UserForm</param>
+        /// <returns>string class progID of the specified control on a UserForm, e.g. "MSForms.CommandButton"</returns>
+        public static string GetDocumentClassControlType(TypeLibWrapper projectTypeLib, string documentClassName, string controlName)
+        {
+            return GetDocumentClassControlType(projectTypeLib.TypeInfos.Get(documentClassName), controlName);
+        }
+
+        /// <summary>
+        /// Returns the class progID of a control on a UserForm
+        /// </summary>
+        /// <param name="ide">Safe-com wrapper representing the UserForm VBA component</param>
+        /// <param name="controlName">Control name, as declared on the UserForm</param>
+        /// <returns>string class progID of the specified control on a UserForm, e.g. "MSForms.CommandButton"</returns>
+        public static string GetDocumentClassControlType(IVBComponent component, string controlName)
+        {
+            return GetDocumentClassControlType(component.ParentProject, component.Name, controlName);
+        }
+
+        /// <summary>
+        /// Returns the class progID of a control on a UserForm
+        /// </summary>
+        /// <param name="documentClass">Low-level ITypeLib wrapper representing the UserForm VBA component</param>
+        /// <param name="controlName">Control name, as declared on the UserForm</param>
+        /// <returns>string class progID of the specified control on a UserForm, e.g. "MSForms.CommandButton"</returns>
+        public static string GetDocumentClassControlType(TypeInfoWrapper documentClass, string controlName)
+        {
+            return documentClass.GetSafeImplementedTypeInfo(0).GetControlType(controlName).GetProgID();
+        }
+        
+        /// <summary>
+        /// Retreives the TYPEFLAGS of a VBA component (e.g. module/class), providing flags like TYPEFLAG_FCANCREATE, TYPEFLAG_FPREDECLID
+        /// </summary>
+        /// <param name="ide">Safe-com wrapper representing the VBE</param>
+        /// <param name="projectName">The VBA project name</param>
+        /// <param name="componentName">The name of the component (module/class etc) to get flags for</param>
+        /// <returns>bool indicating success/failure.</returns>
```

Comment:
I think this `returns` element is wrong. Probably need to update.

Reply:
sorted, thanks.